andela/elven-ah
# #159456828 Local login with email/username and password

#### What does this PR do?
Add ability to login a user locally using email or username and password
#### Description of Task to be completed?
- add support for logging in with email and password
- add support for logging in with username and password
- change the status code and message for the fallback route
#### How should this be manually tested?
- Clone the repo: Run `git clone https://github.com/andela/elven-ah.git` in the folder where you want the repo to be saved.
- Navigate into the newly created folder and install the dependencies: `cd elven-ah && npm install`
- Copy .env.example to .env: `cp .env.example .env`
- Provision a PostgreSQL database and add the credentials to the .env file. 
- Start the server using `npm start`
- Make a POST request to http://localhost:3000/api/auth/signup using postman with your email and verify the email sent to your email.
- Make a POST request to http://localhost:3000/api/auth/login with
    - username or email and
    - password

#### What are the relevant pivotal tracker stories? 
[#159456828](https://www.pivotaltracker.com/story/show/159456828)
